tests/utils_tests/test_baseconv.py

```diff
@@ -1,50 +1,71 @@
-from unittest import TestCase
+from django.test import SimpleTestCase, ignore_warnings
+from django.utils.deprecation import RemovedInDjango50Warning
 
-from django.utils.baseconv import (
-    BaseConverter, base2, base16, base36, base56, base62, base64,
-)
 
-
-class TestBaseConv(TestCase):
+@ignore_warnings(category=RemovedInDjango50Warning)
+class TestBaseConv(SimpleTestCase):
 
     def test_baseconv(self):
+        from django.utils.baseconv import (
+            base2, base16, base36, base56, base62, base64,
+        )
+
         nums = [-10 ** 10, 10 ** 10, *range(-100, 100)]
         for converter in [base2, base16, base36, base56, base62, base64]:
             for i in nums:
                 self.assertEqual(i, converter.decode(converter.encode(i)))
 
     def test_base11(self):
+        from django.utils.baseconv import BaseConverter
+
         base11 = BaseConverter('0123456789-', sign='$')
         self.assertEqual(base11.encode(1234), '-22')
         self.assertEqual(base11.decode('-22'), 1234)
         self.assertEqual(base11.encode(-1234), '$-22')
         self.assertEqual(base11.decode('$-22'), -1234)
 
     def test_base20(self):
+        from django.utils.baseconv import BaseConverter
+
         base20 = BaseConverter('0123456789abcdefghij')
         self.assertEqual(base20.encode(1234), '31e')
         self.assertEqual(base20.decode('31e'), 1234)
         self.assertEqual(base20.encode(-1234), '-31e')
         self.assertEqual(base20.decode('-31e'), -1234)
 
     def test_base64(self):
+        from django.utils.baseconv import base64
+
         self.assertEqual(base64.encode(1234), 'JI')
         self.assertEqual(base64.decode('JI'), 1234)
         self.assertEqual(base64.encode(-1234), '$JI')
         self.assertEqual(base64.decode('$JI'), -1234)
 
     def test_base7(self):
+        from django.utils.baseconv import BaseConverter
+
         base7 = BaseConverter('cjdhel3', sign='g')
         self.assertEqual(base7.encode(1234), 'hejd')
         self.assertEqual(base7.decode('hejd'), 1234)
         self.assertEqual(base7.encode(-1234), 'ghejd')
         self.assertEqual(base7.decode('ghejd'), -1234)
 
     def test_exception(self):
+        from django.utils.baseconv import BaseConverter
+
         with self.assertRaises(ValueError):
             BaseConverter('abc', sign='a')
         self.assertIsInstance(BaseConverter('abc', sign='d'), BaseConverter)
 
     def test_repr(self):
+        from django.utils.baseconv import BaseConverter
+
         base7 = BaseConverter('cjdhel3', sign='g')
         self.assertEqual(repr(base7), '<BaseConverter: base7 (cjdhel3)>')
+
+
+class TestBaseConvDeprecationWarning(SimpleTestCase):
+    def test_deprecation_warning(self):
+        msg = 'The baseconv module is deprecated.'
+        with self.assertRaisesMessage(RemovedInDjango50Warning, msg):
+            from django.utils.baseconv import BaseConverter  # NOQA
```

Comment:
@felixxm Do you have any idea why this test fails with `AssertionError: RemovedInDjango50Warning not raised`?
It passes in my local setup and `django-docker-box`.